OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseDatesPageFragment.kt

```diff
@@ -284,26 +300,28 @@ class CourseDatesPageFragment : OfflineSupportBaseFragment(), BaseFragment.Permi
     }
 
     private fun insertCalendarEvent() {
-        var calendarId: Long = CalendarUtils.getCalendarId(context = contextOrThrow, accountName = accountName, courseName = calendarTitle)
-        calendarId = if (calendarId == (-1).toLong()) {
-            CalendarUtils.createCalendar(context = contextOrThrow, accountName = accountName, courseName = calendarTitle)
-        } else {
-            CalendarUtils.deleteCalendar(context = contextOrThrow, calendarId = calendarId)
-            CalendarUtils.createCalendar(context = contextOrThrow, accountName = accountName, courseName = calendarTitle)
-        }
+        var calendarId: Long = CalendarUtils.createOrUpdateCalendar(context = contextOrThrow, accountName = accountName, calendarTitle = calendarTitle)
         // if app unable to create the Calendar for the course
         if (calendarId == (-1).toLong()) {
             Toast.makeText(contextOrThrow, "Error Adding Calendar, Please try later", Toast.LENGTH_SHORT).show()
             binding.switchSync.isChecked = false
             return
         }
+        addOrUpdateEventsInCalendar(calendarId, false)
+    }
 
+    private fun addOrUpdateEventsInCalendar(calendarId: Long, updateEvents: Boolean) {
```

Comment:
better to delete old events, otherwise the calendar app shows multiple entries against events. like
```kotlin
if (updateEvents) {
            CalendarUtils.deleteAllCalendarEvents(
                context = requireContext(),
                calendarId = calendarId
            )
        }
```